react/features/subtitles/components/web/Captions.tsx

```diff
@@ -1,25 +1,73 @@
+import { Theme } from '@mui/material';
 import React, { ReactElement } from 'react';
 import { connect } from 'react-redux';
+import { withStyles } from 'tss-react/mui';
 
 import { IReduxState } from '../../../app/types';
 import { getLocalParticipant } from '../../../base/participants/functions';
 import { getLargeVideoParticipant } from '../../../large-video/functions';
 import { isLayoutTileView } from '../../../video-layout/functions.web';
+import { calculateSubtitlesFontSize } from '../../functions.web';
 import {
     AbstractCaptions,
     type IAbstractCaptionsProps,
     _abstractMapStateToProps
 } from '../AbstractCaptions';
 
-
 interface IProps extends IAbstractCaptionsProps {
 
+    /**
+     * The height of the visible area.
+     */
+    _clientHeight?: number;
+
     /**
      * Whether the subtitles container is lifted above the invite box.
      */
     _isLifted: boolean | undefined;
+
+    /**
+     * An object containing the CSS classes.
+     */
+    classes?: Partial<Record<keyof ReturnType<typeof styles>, string>>;
 }
 
+
+const styles = (_theme: Theme, props: IProps) => {
+    const { _isLifted = false, _clientHeight } = props;
+
+    const toolbarSize = 48;
+    const fontSize = calculateSubtitlesFontSize(_clientHeight);
+
+    return {
+        transcriptionSubtitles: {
+            bottom: _isLifted ? `${toolbarSize + 36 + 40}px` : `${toolbarSize + 40}px`,
+            fontSize: `${fontSize}px`,
+            left: '50%',
+            maxWidth: '50vw',
+            overflowWrap: 'break-word' as const,
+            pointerEvents: 'none' as const,
+            position: 'absolute' as const,
+            textShadow: `
+                0px 0px 1px rgba(0,0,0,0.3),
+                0px 1px 1px rgba(0,0,0,0.3),
+                1px 0px 1px rgba(0,0,0,0.3),
+                0px 0px 1px rgba(0,0,0,0.3)`,
+            transform: 'translateX(-50%)',
+            zIndex: 7,
```

Comment:
Why 7? Is it relative to something? A comment explaining woould help.

Reply:
Probably. Now the thing is this is a copy paste of the css :)

Reply:
Do you insist on reviewing all z-indexes so that we can write what do we want to cover?

If we are doing this wouldn't it be more beneficial, to review all zIndex levels,  to define all z-index levels as variables with meaningful names and comments?

Reply:
Added a comment for now